src/factory-in-proc.js

```diff
@@ -65,10 +65,14 @@ class FactoryInProc {
       callback = options
       options = {}
     }
-    const IPFS = this.exec
-    new IPFS(options).version((err, _version) => {
-      if (err) { callback(err) }
-      callback(null, _version)
+
+    const IPFS = this.exec // otherwise lint barfs
+    const ipfs = new IPFS(options)
+    ipfs.once('ready', () => {
+      ipfs.version((err, _version) => {
+        if (err) { return callback(err) }
+        callback(null, _version)
+      })
```

Comment:
Is this starting the node? I'm sure this is causing repo lock just to check the version.

Reply:
Hey @dryajov, curious about this as [there is a TODO](https://github.com/ipfs/js-ipfsd-ctl/blob/master/src/factory-daemon.js#L66) to fix this on the go/js factory. Did the mentioned solution hit roadblocks originally? Can we make an issue and work on it separate from this PR?